File: backend/app/models/generation_task.py
"""生成任务模型"""

from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.sql import func
from app.database import Base


class GenerationTask(Base):
    """生成任务表"""
    __tablename__ = "generation_tasks"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    prompt = Column(Text, nullable=False)
    model_name = Column(String(100), nullable=False)
    status = Column(String(20), default="pending", index=True)  # pending, processing, completed, failed
    result_id = Column(Integer, ForeignKey("results.id"), nullable=True)
    error_message = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), index=True)
    completed_at = Column(DateTime(timezone=True), nullable=True)
    
    def __repr__(self):
        return f"<GenerationTask(id={self.id}, user_id={self.user_id}, status={self.status})>"

File: frontend/lib/models/generation_task.dart
/// 生成任务模型
class GenerationTask {
  final int id;
  final int userId;
  final String prompt;
  final String modelName;
  final String status;
  final int? resultId;
  final String? errorMessage;
  final DateTime createdAt;
  final DateTime? completedAt;

  GenerationTask({
    required this.id,
    required this.userId,
    required this.prompt,
    required this.modelName,
    required this.status,
    this.resultId,
    this.errorMessage,
    required this.createdAt,
    this.completedAt,
  });

  factory GenerationTask.fromJson(Map<String, dynamic> json) {
    return GenerationTask(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      prompt: json['prompt'] as String,
      modelName: json['model_name'] as String,
      status: json['status'] as String,
      resultId: json['result_id'] as int?,
      errorMessage: json['error_message'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
      completedAt: json['completed_at'] != null
          ? DateTime.parse(json['completed_at'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'prompt': prompt,
      'model_name': modelName,
      'status': status,
      'result_id': resultId,
      'error_message': errorMessage,
      'created_at': createdAt.toIso8601String(),
      'completed_at': completedAt?.toIso8601String(),
    };
  }

  bool get isPending => status == 'pending';
  bool get isProcessing => status == 'processing';
  bool get isCompleted => status == 'completed';
  bool get isFailed => status == 'failed';
}

